kornia/geometry/liegroup/se3.py

```diff
@@ -0,0 +1,217 @@
+        return concatenate((t, omega), -1)
+
+    @staticmethod
+    def hat(v) -> Tensor:
+        """Converts elements from vector space to lie algebra. Returns matrix of shape :math:`(B,4,4)`.
+
+        Args:
+            v: vector of shape :math:`(B,6)`.
+
+        Example:
+            >>> v = torch.ones((1,6))
+            >>> m = Se3.hat(v)
+            >>> m
+            tensor([[[ 0.,  0., -1.,  1.],
+                     [ 0.,  1.,  0., -1.],
+                     [ 0., -1.,  1.,  0.],
+                     [ 0.,  1.,  1.,  1.]]])
+        """
+        KORNIA_CHECK_SHAPE(v, ["B", "6"])
+        t = v[..., 0:3].reshape(-1, 1, 3)
+        omega = v[..., 3:]
+        rt = concatenate((So3.hat(omega), t.reshape(-1, 1, 3)), 1)
+        # TODO: infer device, dtype
+        return concatenate((zeros(v.shape[0], 4, 1).to(rt.device, rt.dtype), rt), -1)
+
+    @staticmethod
+    def vee(omega) -> Tensor:
+        """Converts elements from lie algebra to vector space.
+
+        Args:
+            omega: 4x4-matrix representing lie algebra of shape :math:`(B,4,4)`.
+
+        Returns:
+            vector of shape :math:`(B,6)`.
+
+        Example:
+            >>> v = torch.ones((1,6))
+            >>> omega_hat = Se3.hat(v)
+            >>> Se3.vee(omega_hat)
+            tensor([[1., 1., 1., 1., 1., 1.]])
+        """
+        KORNIA_CHECK_SHAPE(omega, ["B", "4", "4"])
+        t = omega[..., 3, 1:]
+        v = So3.vee(omega[..., 0:3, 1:])
+        return concatenate((t, v), 1)
+
+    @classmethod
+    def identity(cls, batch_size: int, device=None, dtype=None) -> 'Se3':
+        """Create a Se3 group representing an identity rotation and zero translation.
+
+        Args:
+            batch_size: the batch size of the underlying data.
+
+        Example:
+            >>> s = Se3.identity(batch_size=2)
+            >>> s
+            rotation: real: tensor([[1.],
+                    [1.]], grad_fn=<SliceBackward0>)
+            vec: tensor([[0., 0., 0.],
+                    [0., 0., 0.]], grad_fn=<SliceBackward0>)
+            translation: tensor([[0., 0., 0.],
+                    [0., 0., 0.]])
+        """
+        t: Tensor = as_tensor([0.0, 0.0, 0.0])
+        t = t.repeat(batch_size, 1)
+        return cls(So3.identity(batch_size, device, dtype), t.to(device, dtype))
+
+    def matrix(self) -> Tensor:
+        """Returns the matrix representation of shape :math:`(B, 4, 4)`.
+
+        Example:
+            >>> s = Se3(So3.identity(batch_size=1), torch.ones((1,3)))
+            >>> s.matrix()
+            tensor([[[1., 0., 0., 1.],
+                     [0., 1., 0., 1.],
+                     [0., 0., 1., 1.],
+                     [0., 0., 0., 1.]]], grad_fn=<CatBackward0>)
+        """
+        rt = concatenate((self.r.matrix(), self.t.reshape(-1, 3, 1)), 2)
+        return concatenate(
+            (rt, Tensor([0.0, 0.0, 0.0, 1.0]).repeat([self.t.shape[0], 1, 1]).to(rt.device, rt.dtype)), 1
```

Comment:
```suggestion
            (rt, Tensor([[0.0, 0.0, 0.0, 1.0] for _ in range(self.t.shape[0])])), 1
```
not sure which is faster or more efficient but this is more python. Worth to measure /cc @t-vi any idea here ?

Reply:
i tested both using timeit, performance seems about the same.